### DISCUSSION TITLE
Code Generator: Get all properties (including inherited) of a class using roslyn

### DISCUSSION THREAD

**Discussion by sergioudi** [dotnet/roslyn#59018] [Q&A]
Good day, please, I am trying to autogenerate the viewmodel for my models. But the inherited properties of the model are not retrieved.

How can I get the properties of the inherited classes?

Here is what I've tried

`using Microsoft.CodeAnalysis;
using Microsoft.CodeAnalysis.CSharp.Syntax;
using Microsoft.CodeAnalysis.CSharp;
using Microsoft.CodeAnalysis.Text;
sing System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
namespace Generators
{

    [Generator]
    public class MappedDtoGenerator : ISourceGenerator
    {
        public void Execute(GeneratorExecutionContext context)
        {
            var codeBuilder = new StringBuilder();
            var syntaxTrees = context.Compilation.SyntaxTrees;
            var typeNodes = syntaxTrees.Where(x => (x.GetText().ToString()
                                                           .Contains("[GenerateMappedDto]")))
                                                           .Select(x => x.GetRoot().DescendantNodes().OfType<ClassDeclarationSyntax>().First());

            foreach (var typeNode in typeNodes)
            {
                var typeNodeSymbol = context.Compilation
                                                       .GetSemanticModel(typeNode.SyntaxTree) 
                                                       .GetDeclaredSymbol(typeNode);
                if (typeNodeSymbol == null) {
                    context.Warning(
                        $"Type {typeNode.Identifier.ValueText} does not have a corresponding ISymbol! " +
                        $"Skipping it!");
                }
                else
                {
                    var entityClassNamespace = typeNodeSymbol.ContainingNamespace?.ToDisplayString() ?? "NoNamespace";
                    var generatedDtoClassName = $"{typeNodeSymbol.Name}Dto";
                    var allProperties = typeNode.Members.OfType<PropertyDeclarationSyntax>();
                }
            }
        }
    }
}
`

**Comment by Youssef1313** [comment]
You're relying on the syntax for getting the properties, that's why you don't get anything inherited.

In fact, even if inheritance isn't considered, your approach will not work for partial declarations of a class. Only one of the partial will have the attribute and properties in the other partial will not be considered.

Have you tried getting the properties through `typeNodeSymbol`?

**Reply by sergioudi** [reply]
I tried using typeNodeSymbol, but it also only returns the properties of the Model class, the inherited properties are not returned.

**Reply by Youssef1313** [reply]
Try to walk through `BaseType`s
